src/model/kaniRunner.ts

```diff
@@ -0,0 +1,185 @@
+				} else {
+					reject(new Error(`Invalid kani path: ${cargoPath}`));
+				}
+			} catch (err) {
+				reject(err);
+			}
+		});
+	});
+}
+
+/**
+ * Function that runs cargo kani [args]
+ *
+ * @param kaniCommand - Full sanitized command created by kaniCommandCreate module
+ * @returns the path for the binary cargo-kani (either the installed binary or the development one)
+ */
+export async function runKaniCommand(
+	harnessCommand: string,
+	cargoKaniMode: boolean = false,
+): Promise<any> {
+	// get the full resolved path for the root directory of the crate
+	const directory = path.resolve(getRootDir());
+	const commmandSplit: CommandArgs = splitCommand(harnessCommand);
+
+	// Get cargo command and args for the command to be executed
+	const command = commmandSplit.commandPath;
+	const args = commmandSplit.args;
+	let kaniBinaryPath = '';
+
+	if (command == 'cargo' || command == 'cargo kani') {
+		const kaniBinaryPath = await getKaniPath('cargo-kani');
+		const options = {
+			shell: false,
+			cwd: directory,
+		};
+
+		return executeKaniProcess(kaniBinaryPath, args, options, cargoKaniMode);
+	} else if (command == 'kani') {
+		kaniBinaryPath = await getKaniPath(command);
+		const options = {
+			shell: false,
+		};
+
+		return executeKaniProcess(kaniBinaryPath, args, options, cargoKaniMode);
+	} else {
+		return false;
+	}
+}
+
+/**
+ * Function that returns the diff message to be displayed
+ *
+ * @param command - Full sanitized command created by kaniCommandCreate module
+ * @returns the path for the binary cargo-kani (either the installed binary or the development one)
+ */ export async function createFailedDiffMessage(command: string): Promise<KaniResponse> {
+	// Root dir of the crate and the command and args to be executed
+	const directory = path.resolve(getRootDir());
+	const commmandSplit: CommandArgs = splitCommand(command);
+
+	// Get the args for the kani command to run
+	const args = commmandSplit.args;
+
+	// Check the command running and execute that with the full path and safe options
+	if (commmandSplit.commandPath == 'cargo' || commmandSplit.commandPath == 'cargo kani') {
+		const kaniBinaryPath = await getKaniPath('cargo-kani');
+		const options = {
+			shell: false,
+			cwd: directory,
+		};
+
+		return new Promise((resolve, reject) => {
+			execFile(kaniBinaryPath, args, options, (error, stdout, stderr) => {
+				if (stdout) {
+					const responseObject: KaniResponse = responseParserInterface(stdout);
+					resolve(responseObject);
+				}
+			});
+		});
+	} else if (commmandSplit.commandPath == 'kani') {
+		const kaniBinaryPath = await getKaniPath('kani');
+		const options = {
+			shell: false,
+		};
+
+		return new Promise((resolve, reject) => {
+			execFile(kaniBinaryPath, args, options, (error, stdout, stderr) => {
+				if (stdout) {
+					const responseObject: KaniResponse = responseParserInterface(stdout);
+					resolve(responseObject);
+				}
+			});
+		});
+	} else {
+		// Error Case
```

Comment:
Would be nice to clarify the errors you're expecting here.

Reply:
Will be addressing this in a future PR - https://github.com/model-checking/kani-vscode-extension/issues/45. There's a lot to consider and design around and I want to keep this PR specific to the security improvement.